src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts

```diff
@@ -1160,15 +1172,11 @@ var requirejs = (function() {
 			return;
 		}
 
-		// TODO: use proper handler
-		const p = new Promise<void>(resolve => {
-			const sub = this.webview?.onMessage(e => {
-				if (e.message.type === 'initializedMarkup') {
-					resolve();
-					sub?.dispose();
-				}
-			});
-		});
+		this.firstInit = false;
```

Comment:
`this.firstInit` is set to `false` when we request viewport warmup. Then when the webview preloads sends back `initialized`event, we will run `initializeWebViewState`. Since `this.firstInit` is set to `false`, it will trigger `initializeMarkup` again, which will `complete` the warmup promise immediately.

I think we want to make sure that we only send requests after `initialized` event come back, then only one single request to `intializeMarkup` for viewport is sent.